File: typescipt/src/language/expressions/parseExpressionFunctionsResult.ts


public sealed class ParseExpressionFunctionsResult : ParseResult<ExpressionFunction> {
   private ParseExpressionFunctionsResult(ExpressionFunction result) : base(result) {
   }

   private ParseExpressionFunctionsResult(boolean success, string errorMessage) : base(success, errorMessage) {
   }

   public static success(result: ExpressionFunction =: ExpressionFunction null: ExpressionFunction): ParseExpressionFunctionsResult {
     return new ParseExpressionFunctionsResult(result);
   }

   public static failed(errorMessage: string): ParseExpressionFunctionsResult {
     return new ParseExpressionFunctionsResult(false, errorMessage);
   }
}

File: src/Lexy.Compiler/Language/Expressions/ParseExpressionFunctionsResult.cs
using Lexy.Compiler.Language.Expressions.Functions;

namespace Lexy.Compiler.Language.Expressions;

public sealed class ParseExpressionFunctionsResult : ParseResult<FunctionCallExpression>
{
    private ParseExpressionFunctionsResult(FunctionCallExpression result) : base(result)
    {
    }

    private ParseExpressionFunctionsResult(bool success, string errorMessage) : base(success, errorMessage)
    {
    }

    public static ParseExpressionFunctionsResult Success(FunctionCallExpression result = null)
    {
        return new ParseExpressionFunctionsResult(result);
    }

    public static ParseExpressionFunctionsResult Failed(string errorMessage)
    {
        return new ParseExpressionFunctionsResult(false, errorMessage);
    }
}